qualtran/bloqs/chemistry/trotter/trotterized_unitary.py

```diff
@@ -0,0 +1,121 @@
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Bloq for building a Trotterized unitary"""
+
+from functools import cached_property
+from typing import Dict, Iterable, Union
+
+import attrs
+import sympy
+
+from qualtran import Bloq, bloq_example, BloqBuilder, BloqDocSpec, Signature, Soquet, SoquetT
+
+
+@attrs.frozen
+class TrotterizedUnitary(Bloq):
+    r"""Implement arbitrary trotterized unitary given any Trotter splitting of the Hamiltonian.
+
+    Given an arbitrary splitting of the Hamiltonian into $\Gamma$ terms
+
+    $$
+        H = \sum_{\gamma=1}^\Gamma H_\gamma
+    $$
+
+    then the unitary $e^{-i t H}$ can be approximately implemented via a $p$-th order product
+    formula
+
+    $$
+        S_p(t) = \prod_{v=1}^{\Upsilon}\prod_{k=1}^\Gamma e^{-it a_{v,\gamma} H_{\pi_v(\gamma)}}
+    $$
+
+    where $\Upsilon$ is the number of `stages`, $a_{v, \gamma}$ are real numbers
+    and $\pi_v(\gamma)$ is a permutation of the Hamiltonian term labels.
+
+    To construct the unitary we adopt the convention from the second reference
+    which simply requires a list of the coefficients and a list of indices which
+    index the hamiltonian terms.
+
+    For example, the second order Suzuki splitting would have indicies = (0, 1, 0)
+    and coeffs = (0.5, 1, 0.5), which would build
+
+    $$
+        e^{-i \frac{t}{2} H_0} e^{-i t H_1} e^{-i \frac{t}{2} H_0}
+    $$
+
+    Args:
+        bloqs: A tuple of bloqs of length $\Gamma$ which implement the unitaries for
+            each term in the Hamiltonian.
+        indices: A tuple of integers which specifies which bloq to apply when
+            forming the unitary as a product of unitaries.
+        coeffs: The coefficients $a$ which appear in the expression for the unitary.
+        timestep: The timestep $t$.
+
+    Registers:
+        system: The system register to which to apply the unitary.
+
+    References:
+        [Theory of Trotter Error with Commutator Scaling](
+            https://journals.aps.org/prx/abstract/10.1103/PhysRevX.11.011020) Eq. 12 page 7.
+
+        [Trotter error with commutator scaling for the Fermi-Hubbard model](
+            https://arxiv.org/abs/2306.10603) see github repo for software to produce splittings.
+    """
+
+    bloqs: Iterable[Bloq]
+    indices: Iterable[int]
+    coeffs: Iterable[Union[float, sympy.Expr]]
+    timestep: Union[float, sympy.Expr]
+
+    def __attrs_post_init__(self):
+        ref_sig = self.bloqs[0].signature
+        for bloq in self.bloqs[1:]:
+            assert bloq.signature == ref_sig
+
+    @cached_property
+    def signature(self) -> Signature:
+        return self.bloqs[0].signature
+
+    def build_composite_bloq(self, bb: 'BloqBuilder', **soqs: SoquetT) -> Dict[str, 'Soquet']:
+        for i, c in zip(self.indices, self.coeffs):
```

Comment:
nit: Let's use `a` instead of `c` since that's what we used in the docstring
```suggestion
        for i, a in zip(self.indices, self.coeffs):
```

Reply:
Done.